apps/spotrem/app.js

```diff
@@ -98,25 +98,39 @@ let swipeHandler = function(LR, _) {
   }
 };
 
+let buttonHandler = (n)=>{load();};
+
 // Navigation input on the main layout
 let setUI = function() {
 // Bangle.setUI code from rigrig's smessages app for volume control: https://git.tubul.net/rigrig/BangleApps/src/branch/personal/apps/smessages/app.js
-  Bangle.setUI(
-    {mode : "updown",
+  Bangle.setUI({
+      mode : "updown",
       remove : ()=>{
-        Bangle.removeListener("touch", touchHandler);
-        Bangle.removeListener("swipe", swipeHandler);
-        clearWatch(buttonHandler);
-        widgetUtils.show();
-      }
+       widgetUtils.show();
+      },
+      /**
+       * TODO: Proposal: Handlers added at the same level as the `mode : "updown"` entry
+       * will replace those that come with the mode - if there were any of that type.
+       * The handlers in the `extra` object below will still be added in addition to these ones.
+       **/
+      swipe : swipeHandler,
+      btn : buttonHandler,
+      btnEdge : "rising", // TODO: Proposal: a new, optional, way to choose what edge the button(-s) should act on.
```

Comment:
@gfwilliams what do you think of the idea to be able to specify the desired edge like this or some other way?

Reply:
Well, I was going to say this didn't feel great but I couldn't think of a better way...

But then I was thinking: "What if you want to do different things on Rising *and* falling edges?"

So what if we had `btn` (as before) but then `btnPress`/`btnRelease` callbacks could be specified? That feels like maybe a tidier solution? What do you think?

Reply:
... the only downside of this is I guess it's not easily backwards compatible - but if we want this stuff to work outside of the `mode:custom` then maybe that doesn't matter as it wouldn't work now anyway

Reply:
I think I agree with your callbacks solution being tidier.

But just regarding this: 
> But then I was thinking: "What if you want to do different things on Rising and falling edges?"

I'm not saying it's the better solution, just another one. If "both" was specified in my example above that could be passed on to the `setWatch` call inside setUI and then the `buttonHandler` function would be responsible for handling both edges.

If you feel callbacks are better that carries much weight for me :)

Reply:
@gfwilliams 

I think I agree with your callbacks solution being tidier.

But just regarding this: 
> But then I was thinking: "What if you want to do different things on Rising and falling edges?"

I'm not saying it's the better solution, just another one. If "both" was specified in my example above that could be passed on to the `setWatch` call inside setUI and then the `buttonHandler` function would be responsible for handling both edges.

And if you feel callbacks are better that carries much weight for me :)